packages/blocks/src/api/parser/convert-legacy-block.js

```diff
@@ -88,25 +88,41 @@ export function convertLegacyBlockNameAndAttributes( name, attributes ) {
 			( name === 'core/paragraph' ||
 				name === 'core/heading' ||
 				name === 'core/image' ||
-				name === 'core/button' )
+				name === 'core/button' ) &&
+			newAttributes.metadata.bindings.__default?.source !==
+				'core/pattern-overrides'
 		) {
 			const bindings = [
 				'content',
 				'url',
 				'title',
+				'id',
```

Comment:
To do this strictly correctly, we should use `PARTIAL_SYNCING_SUPPORTED_BLOCKS` to match attributes to specific block types. Right now, this function is duplicating what is already some temporary hardcoded information, which just seems a bit too error-prone.

That said, IMO the strictly correct way seems overkill for the purpose of this branch. We could just loop over all keys of `.metadata.bindings` to replace `.source.name` and call it a day.

Reply:
I agree we should probably just iterate through the bindings and remove the ones with source `core/pattern-overrides`, and avoid having this hardcoded list.